tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/Helpers/TelemetryHelper.cs

```diff
@@ -165,9 +165,45 @@ _ when data.TryGetPayload<AppIntegrationsChangedPayload>(TelemetryRequestTypes.A
                 }
             }
 
+            var spansCreatedByIntegration = new ConcurrentDictionary<string, MetricData>();
+            var metricsPayloads =
+                allData
+                    .Select(
+                        data => data switch
+                        {
+                            _ when data.TryGetPayload<GenerateMetricsPayload>(TelemetryRequestTypes.GenerateMetrics) is { } p => p.Series.Where(s => s.Metric == "spans_created"),
+                            _ => null,
+                        })
+                    .Where(x => x is not null);
+
+            // Flatten the spans_created metrics
+            foreach (var metricPayload in metricsPayloads)
+            {
+                foreach (var metricEntry in metricPayload)
+                {
+                    spansCreatedByIntegration.TryAdd(metricEntry.Tags.First(s => s.StartsWith("integration_name:")), metricEntry);
+                }
+            }
+
+            var integrationName = integrationId.ToString();
+            var integrationNameTagValue = integrationName switch
+            {
+                "OpenTelemetry" => "otel",
+                _ => integrationName.ToLowerInvariant(),
+            };
+
+            if (enabled)
+            {
+                spansCreatedByIntegration.Should().NotBeEmpty();
+
+                var spansCreated = spansCreatedByIntegration.Should().ContainKey($"integration_name:{integrationNameTagValue}").WhoseValue;
```

Comment:
This adds a new check (for all our integration tests!) that when the integration is enabled we also receive a `spans_created` metric that corresponds to the enabled integration.

The "otel" check above shows how this is the only integration whose tag value differs from its corresponding `IntegrationId.ToString()` 😅